packages/core/src/rules/common/assertions/index.ts

```diff
@@ -15,11 +15,10 @@ export const Assertions: Oas3Rule | Oas2Rule = (opts: object) => {
   );
 
   for (const [index, assertion] of assertions.entries()) {
-    const assertId =
+    assertion.assertId =
       (assertion.assertionId && `${assertion.assertionId} assertion`) || `assertion #${index + 1}`;
-
     if (!assertion.subject) {
-      throw new Error(`${assertId}: 'subject' is required`);
+      throw new Error(`${assertion.assertId}: 'subject' is required`);
```

Comment:
I'd avoid this mutation. If we throw an error, the `assertions` variable will get half-mutated. This could potentially lead to errors that would be hard to detect.